pkg/trace/obfuscate/sql.go

```diff
@@ -117,7 +117,7 @@ func (f *replaceFilter) Filter(token, lastToken TokenKind, buffer []byte) (token
 	case '?':
 		// Cases like 'ARRAY [ ?, ? ]' should be collapsed into 'ARRAY [ ? ]'
 		return markFilteredGroupable(token), questionMark, nil
-	case TableName:
+	case TableName, ID:
```

Comment:
Hmm I think this is overly broad. column names with numbers will also have the quantization applied to them (as well as any other token). Can you add a few more test cases and see the result?

Reply:
Yeah, this does make it too broad. Here are two cases and the their results. This quantizes numbers outside of the table, which isn't the behavior we want out of this option. Going to investigate to see how complex it would be to get the obfuscator to interpret the token correctly.

```
Before: SELECT ddu1.name, ddo2.id, ddk3.app_key FROM dd3120.user ddu1, dd1931.orgs ddo2, dd53819.keys ddk3
After: SELECT ddu?.name, ddo?.id, ddk?.app_key FROM dd?.user ddu?, dd?.orgs ddo?, dd?.keys ddk?
```

```
Before: SELECT articles.* FROM articles19238 WHERE articles.id5 = 1 LIMIT 1, 5
After: SELECT articles.* FROM articles? WHERE articles.id? = ? LIMIT ?
```